cranelift/filetests/filetests/runtests/bitcast-ref64.clif

```diff
@@ -0,0 +1,26 @@
+test run
+target aarch64
+; the interpreter, x86_64, and s390x do not support bitcasting to/from
+; references
+
+function %bitcast_ir64(i64) -> b1 {
+block0(v0: i64):
+  v1 = bitcast.r64 v0
```

Comment:
I thought there were some issues with this: https://github.com/bytecodealliance/wasmtime/issues/3217 ?

Reply:
Hm, I was not aware of that issue. Looking at it, I _think_ this is fine since we're not dealing with precise references to an actual value - merely arbitrary integer values which are treated as references and passed into `is_null`. (Unless, of course, there's something I'm not aware of...)

Reply:
The thing that concerned me the most was a comment by @bjorn3:

>> I did not attempt to understand why this worked
>
> I am fairly certain it doesn't actually fix the problem, but instead allows a silent miscompilation where the stackmap may be missing entries or have erroneous entries.

In response to someone who tried to do an identical thing (as far as I can tell). So I wonder, do we still have the stackmaps issue on bitcasts? And while this operation is allowed, are we silently allowing bugs?

Reply:
So, at least to my reading,
> I did not attempt to understand why this worked

referred to them 'solving' the issue by "vendoring regalloc and commenting out the assert to not be there" - which I think is what @bjorn3 means: removing the assert does not fix the problem, merely fails to catch the miscompilation.

I'm hopeful that _that_ issue should not be present here, as I have not modified regalloc in such a way (or at all): if the issue was to show up, I myself would expect to see the initial error message detailed in that issue.